### ISSUE TITLE
unexpected error multi-tabs

### ISSUE THREAD

**Issue by bkardell** [brave/browser-laptop#2634]
**Did you search for similar issues before submitting this one?**
Yes

**Describe the issue you encountered:**
I leave brave open all the time, it's my "secondary primary browser" - use it for most of my social media/news reading and I've usually got 2-10 tabs open. At least once a day recently I'll suddenly notice that _several_ tabs (usually all, but not always) will suddenly simultaneously swap into error state).  Sometimes clicking refresh will bring them back to life, other times not - it just hangs. The same url will always open in other browsers.  Restarting brave will also open them back up. 

**Expected behavior:**
Not that :)

- Platform (Win7, 8, 10? macOS? Linux distro?):
Mac OSX 10.9.2

- Brave Version:
0.11.0

- Steps to reproduce:
    I've been monitoring this to see if I can find any rhyme or reason, specific sites, things I'm doing, network status, etc - but I've been unable to notice anything particular. 

- Screenshot if needed:
<img width="1362" alt="screen shot 2016-07-21 at 10 04 54 am" src="https://cloud.githubusercontent.com/assets/870501/17025763/65613ad8-4f2b-11e6-8660-9996ed7e8ff2.png">

- Any related issues:

**Comment by bbondy** [comment]
cc @bridiver any ideas?

**Comment by bridiver** [comment]
are you using any of the password manager extensions?

**Comment by bkardell** [comment]
nay.

**Comment by bbondy** [comment]
Is it the same with 0.11.1?

**Comment by bkardell** [comment]
I'm not really sure when that update occurred unfortunately, but I think yes because I feel like I remember thinking 'oh, maybe this will fix it'.  Sorry I am only submitting it now, I just didn't want to file with no data and each time it happened thus far I was in a position where I couldn't take the time.

**Comment by neeklamy** [comment]
I’m seeing the exact same thing, several tabs will all go bad at the same time.

The various things I’ve tried to reload a broken page:
- Hitting reload (⌘R) or the reload button in the tool bar.
- Focusing the URL bar (⌘L) and hitting <kbd>Enter</kbd>.
- Closing the tab and using the *Reopen Last Closed Tab* menu item.
- Pasting the page URL into an entirely new Brave window…

At that point, relaunching Brave is the only thing that works.

**Comment by bridiver** [comment]
we're investigating a possible memory issue as the cause. If you have any Brave crash reports in `~/Library/Logs/DiagnosticReports` they would be very useful.

**Comment by bkardell** [comment]
@bridiver here's a zip with 584 of them...
[Archive.zip](https://github.com/brave/browser-laptop/files/377206/Archive.zip)

**Comment by bridiver** [comment]
Thanks! Yep, it's the same `v8::internal::V8::FatalProcessOutOfMemory`. Working on it now.

**Comment by bridiver** [comment]
cc @BrendanEich

**Comment by bridiver** [comment]
@bkardell you might also want to file a bug with google for the 20 crashes a day you have from GoogleSoftwareUpdateAgent :)

**Comment by bkardell** [comment]
@bbondy blocked by corp, I think - it's not them, it's me :)

**Comment by bsclifton** [comment]
This also happens to me when opening brave when my internet is down (happens to all tabs though)

**Comment by srirambv** [comment]
I have seen if my internet is down it adds more error tabs(along with the tabs from previous session) when opening the browser.

**Comment by bbondy** [comment]
My wife complained about it spawning new error tabs too btw like the last comment.

**Comment by bridiver** [comment]
the internet errors are definitely a separate issue. This bug is due to the oom referenced above and still working on a fit

**Comment by bbondy** [comment]
I think we've identified OOM In general as the cause and @bridiver is working on various memory profile related things. Some of them already released in 0.11.4.  I don't think we need to track this in an issue since it'll be ongoing work. Please re-open if you disagree though.
